hadoop-tools/hadoop-aws/src/main/java/org/apache/hadoop/fs/s3a/S3AFileSystem.java

```diff
@@ -4366,15 +4378,38 @@ private void requireSelectSupport(final Path source) throws
           InternalConstants.STANDARD_OPENFILE_KEYS,
           "for " + path + " in non-select file I/O");
     }
+    FileStatus status = parameters.getStatus();
```

Comment:
nit: This `status` can be named `providedStatus` or something. Clearer as it's referred multiple times following.

Reply:
will do